lib/trebuchet/backend/redis_hammerspaced.rb

```diff
@@ -0,0 +1,102 @@
+require 'trebuchet/backend/redis'
+require 'json'
+
+class Trebuchet::Backend::RedisHammerspaced < Trebuchet::Backend::Redis
+
+  # This class will rely on a cron job to sync all trebuchet features
+  # to local hammerspace thus this class never directly updates hammerspace
+  # We also cache in memory the features and rely on before_filter
+  # to lazily invalidate local cache
+
+  attr_accessor :namespace
+
+  def initialize(*args)
+    @namespace = 'trebuchet/'
+    begin
+      # args.first must be a hash
+      @options = args.first
+      @redis = @options[:client]
+      @hammerspace = @options[:hammerspace]
+      unless @options[:skip_check]
+        # raise error if not connectedUncaught ReferenceError: google is not defined
+        @redis.exists(feature_names_key) # @redis.info is slow and @redis.client.connected? is NOT reliable
+        @hammerspace.has_key?(feature_names_key)
+      end
+    rescue Exception => e
+      raise Trebuchet::BackendInitializationError, e.message
+    end
+  end
+
+  def get_strategy(feature_name)
+    if cached_strategies.has_key?(feature_name)
+      # use cached if available (even if value is nil)
+      cached_strategies[feature_name]
+    else
+      # call to hammerspace
+      cache_strategy feature_name, get_strategy_hammerspace(feature_name)
+    end
+  end
+
+  def get_strategy_hammerspace(feature_name)
+    # Read from hammerspace
+    return nil unless @hammerspace.has_key?(feature_key(feature_name))
+    # h will be a string, we need to convert it back to Hash
+    h = @hammerspace[feature_key(feature_name)]
+    begin
+      h = JSON.load(h)
+    rescue
+      return nil
+    end
+    unpack_strategy(h)
+  end
+
+  def unpack_strategy(options)
+    # We don't need to further convert values
+    # because it's already taken care of
+    # by the refresh cron job
+    return nil unless options.is_a?(Hash)
+    [].tap do |a|
+      options.each do |k, v|
+        key = k.to_sym
+        a << key
+        a << v
```

Comment:
While I'm not sure I agree with [this comment](https://github.com/airbnb/trebuchet/blob/9db945f1f914413533a93ca5a0635cf9d4082e55/lib/trebuchet/backend/redis.rb#L55) since a hash encodes to JSON just fine, I wonder if we should preserve the format here by doing `v.first` and writing the cron job such that it does `[options].to_json`. Technically it's a small bit of extra overhead, but it preserves compatibility with other backends in case we decide to do something with the array in the future.